apps/mobile/src/app/(home)/settings/wallet/configure/[wallet]/index.tsx

```diff
@@ -75,11 +78,36 @@ function ConfigureWallet({ wallet }: ConfigureWalletProps) {
     return { success: true };
   }
 
-  function removeWallet() {
+  function removeWalletImpl() {
     router.back();
     dispatch(userRemovesWallet({ fingerprint: wallet.fingerprint }));
   }
 
+  async function secureRemoveWallet() {
+    if (securityLevelPreference === 'secure') {
+      const result = await LocalAuthentication.authenticateAsync();
+      if (result.success) {
+        removeWalletImpl();
+      } else {
+        displayToast({
+          title: t({
+            id: 'configure_wallet.delete_wallet.authentication_failed',
+            message: 'Authentication failed',
+          }),
+          type: 'error',
+        });
+      }
+    }
+  }
+
+  async function removeWallet() {
+    if (securityLevelPreference === 'secure') {
+      await secureRemoveWallet();
+    } else {
+      removeWalletImpl();
+    }
+  }
```

Comment:
thoughts on early return alternative?

```suggestion
  async function removeWallet() {
    if (securityLevelPreference === 'secure') {
      await secureRemoveWallet();
      return;
    }
    removeWalletImpl();
  }
```